### PR TITLE
Fix part of #4064: Add new learner analytics events

### PR DESCRIPTION
**Pull request by BenHenning**
## Explanation
Fix part of #4064

This PR integrates the constituent Dagger modules introduced in #4267 broadly in the app so that new logs can be easily added without changing a bunch of files (see #4270).

Some notes on the history of this PR:
- This PR is a rebase of #4258 to remove dependencies on #2173 (which has since been merged into developer) for a much cleaner history.
- This PR is pulling out elements from #4118, #4247, and #4248 which contained completed work by @Sarthak2601.
- This PR introduces new logging that didn't exist in the original PRs (#4118, #4247, and #4248)

This PR broadly adds all of the analytics event described in the TDD above (& its corresponding PRD), needed for an upcoming user study with the app. It *does* enable actual logging if analytics were to be enabled in the Android manifest (even though the UI side isn't implemented yet--that's planned for #4271).

## Essential Checklist
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
N/A -- This PR doesn't make UI changes, and existing flows shouldn't be affected.

### CODE REVIEW

**Comment by BenHenning** [comment]
Still active.

**Comment by BenHenning** [comment]
PTAL @anandwana001 at the whole PR (as you'll be the sole reviewer for this change).

I'm also planning to defer @vinitamurthi's review per earlier conversations.

File: utility/src/test/java/org/oppia/android/util/logging/EventBundleCreatorTest.kt [hunk 1]
```diff
@@ -63,7 +63,7 @@ class EventBundleCreatorTest {
     private const val TEST_SKILL_ID_2 = "test_skill_id_2"
     private const val TEST_SUB_TOPIC_INDEX = 1
     private const val TEST_SUB_TOPIC_INDEX_STR = "1"
-    private const val TEST_LEARNER_ID = "test_learner_id"
+    private const val TEST_LEARNER_ID = "test_ed_ld_learner_id"
```
Comment (anandwana001) [inline utility/src/test/java/org/oppia/android/util/logging/EventBundleCreatorTest.kt:66]: Should we make it more understandable? - `test_ed_ld_learner_id`

**Review by anandwana001 — APPROVED** [review]
LGTM, left a thought.

File: utility/src/test/java/org/oppia/android/util/logging/EventBundleCreatorTest.kt (hunk 1 above)
Comment (BenHenning) [reply utility/src/test/java/org/oppia/android/util/logging/EventBundleCreatorTest.kt:66]: @anandwana001 which part should be made more understandable? If it's the 'ed', 'ld', etc. we unfortunately have a very limited number of characters for event properties, so I was trying to capture context & keep uniqueness without running over that. Per https://support.google.com/firebase/answer/9237506?hl=en we have only 40 characters for these names, so they do need to be really short.

Suggestions for alternatives are welcome. :)

Comment (anandwana001) [reply utility/src/test/java/org/oppia/android/util/logging/EventBundleCreatorTest.kt:66]: Sounds good then as we have only 40 char limit.

**Comment by BenHenning** [comment]
Still active.

**Pull request merged**